test/ArbitraryTest.re

```diff
@@ -175,8 +175,29 @@ describe("complex built-in arbitraries", () => {
         constTrue,
       ),
     );
+  });
+
+  it("json", () => {
+    // Despite the names, these are really *any* JSON value
     FcAssert.sync(property1(Objects.jsonObject(5), eq));
     FcAssert.sync(property1(Objects.unicodeJsonObject(5), eq));
+    let checkProducesJsonValue = arb => {
+      open Js.Json;
+      //let classifiedArb = arb->Derive.map(Js.Json.classify);
+      let checkProduces = kind =>
+        SyncUnit.property1(arb->Derive.filter(v => v->test(kind)), ignore)
```

Comment:
If I break your new test by only producing one type of value, it hangs. It looks like your test needs to use `fc.pre` not `fc.filter` (I realise `.pre` doesn't have a binding at the moment).

> The advantage of `fc.pre(...)` over `.filter(...)` is that runs having too many rejected values will be marked as faulty. When used in combination of `fc.check(...)` it can help to design new filtered arbitraries as the number of skipped values will be computed and available in the output.

https://github.com/dubzzz/fast-check/blob/master/documentation/1-Guides/Tips.md#filter-invalid-combinations-using-pre-conditions